pullrequests/lowest_common_ancester_of_a_binary_search_tree/step2.go

```diff
@@ -0,0 +1,32 @@
+//lint:file-ignore U1000 Ignore all unused code
+package lowestcommonancesterofabinarysearchtree
+
+/*
+より見やすくなるようにリファクタしました。また、再帰を使った実装もしてみました。
+エラー処理についてはStep1と同様です。
+*/
+func lowestCommonAncestorIterative(root, p, q *TreeNode) *TreeNode {
+	node := root
+	for node != nil {
+		if p.Val < node.Val && q.Val < node.Val {
+			node = node.Left
+			continue
+		}
+		if node.Val < p.Val && node.Val < q.Val {
```

Comment:
ループに入る前に p と q の大小関係を比較すると、if 文内の条件が少しシンプルかつわかりやすくなるかもです。

```suggestion
        smaller, larger := p.Val, q.Val
        if p.Val > q.Val:
                smaller, larger := q.Val, p.Val
        for node != nil {
                ...
		if node.Val < smaller:
```

Reply:
ありがとうございます。確かにそうするのも良さそうですね